### DISCUSSION TITLE
Cut and Paste of double quoted characters from Visual Studio Code to DBeaver SQL editor removes chars between quotes

### DISCUSSION THREAD

**Discussion by rotorboy** [dbeaver/dbeaver#11597] [Q&A] [answered]
#### System information: 
- Operating system (distribution) and version: MacOS 10.14
- DBeaver version 7.3.3
- Additional extensions 

#### Connection specification:
- Database name and version: MariaDB
- Driver name: default
- Do you use tunnels or proxies (SSH, SOCKS, etc)? no

#### Describe the problem you're observing:

When a string containing double-quoted characters is copied from VS Code to DBeaver directly, the double quotes and the characters between double quotes get removed.

For example:

CONCAT('{', '"1":"', SUM(f1), '",','"5":"', SUM(f2), '",','"10":"', SUM(f3), '"','}')  

turns into:

 CONCAT('{', ':,',':,',':','}')

Pasting into an intermediary app (eg text or email) prevents this from happening. 

#### Steps to reproduce, if exist:

open a file VS code. create a variable that holds SQL query string containing aforementioned double-quoted characters . Stop in the debugger, and select "copy value" while rolled over string variable. paste into DBeaver SQL editor window.

**Comment by kseniiaguzeeva** [comment, marked as answer]
Looks like DBeaver special feature. Please try to disable option **Extract SQL from source code** in Preferences>Editors>SQL Editor>Code Editor. It should help.
![Screenshot 2021-02-18 at 11 10 08](https://user-images.githubusercontent.com/51405061/108325402-1986b580-71da-11eb-9860-2c61b4e97927.png)
